### ISSUE TITLE
Adding more Learning-Based Repair tools to Cerberus

### ISSUE THREAD

**Issue by Marti2203** [nus-apr/cerberus#32]
There is a list of possible tools one can work on adding to Cerberus in order to increase the breadth of the framework.

* [KNOD](https://github.com/lin-tan/knod)
* [CURE](https://github.com/lin-tan/CURE)
* [CoCoNut](https://github.com/lin-tan/CoCoNut-Artifact)
* [VRepair](https://github.com/SteveKommrusch/VRepair)

**Comment by thanhlecongg** [comment]
Hi @Marti2203, did you find trained models for KNOD and CoCoNUT? I'm working on add more learning-based repair to Cerberus but I cannot find trained models for these techniques.

**Comment by Marti2203** [comment]
Hi @thanhlecongg , I unfortunately have not found CoCoNut's model yet, was searching for it last week. I have to check again for KNOD

**Comment by thanhlecongg** [comment]
Yeah, me too. For KNOD, I also cannot find the models. However, I found there are two other techniques with trained models:
- SelfAPR: [Code](https://github.com/ASSERT-KTH/SelfAPR) [Model](https://zenodo.org/record/7030141)
- AlphaRepair: [Code](https://zenodo.org/record/6819444) (This technique uses CodeBERT directly so it is available in hugging face)

Btw, Do you plan to add any learning-based approaches in the near future? I also plan to do that so we can share to avoid wasting efforts.

**Comment by Marti2203** [comment]
I just checked for KNOD - yeah, I also cannot find the model. Thought that the docker contained it... I am unfortunately not going to add more learning-based tools soon as I am helping for APR-COMP :D Will message you if I am planning to add a tool

**Comment by thanhlecongg** [comment]
Ok. No worries. I also planned to do it. I will contribute after I complete.

**Comment by Marti2203** [comment]
Thanks a lot! :)

**Comment by Marti2203** [comment]
Will add you to the assigned people to allow for easier tracking

**Comment by thanhlecongg** [comment]
AlphaRepair has been added by Pull request #100. I will keep adding more learning-based repair tools.

**Comment by thanhlecongg** [comment]
SelfAPR has been added by Pull request #103.

**Comment by thanhlecongg** [comment]
@Marti2203 I'm working on RewardRepair. Unfortunately, they do not provide preprocessing code for their work. I have tried to reach the authors but have not received their response until now. I have implemented my own preprocessing code following their descriptions in the paper but the processed inputs are not identical to their provided data. Should I make a pull request for RewardRepair?

**Comment by Marti2203** [comment]
You can do a PR, but add an `I` to the name so there is some way of tracking that it is different

**Comment by thanhlecongg** [comment]
Ok, I got it.

**Comment by thanhlecongg** [comment]
RewardRepair has beed added by Pull request #104.

**Comment by thanhlecongg** [comment]
RewardRepair, SelfAPR, and AlphaRepair are all the tools that I can find the trained model until now. So I will pause this task.
